packages/clerk-js/src/core/resources/UserSettings.ts

```diff
@@ -84,6 +86,11 @@ export class UserSettings extends BaseResource implements UserSettingsResource {
           ? defaultMaxPasswordLength
           : Math.min(data?.password_settings?.max_length, defaultMaxPasswordLength),
     };
+    this.usernameSettings = {
+      ...data.username_settings,
+      min_length: Math.max(data?.username_settings?.min_length, 4),
+      max_length: Math.min(data?.username_settings?.max_length, 64),
```

Comment:
Only thing i'm skeptical of is what happens if we decided to change the limits. For example if FAPI in the future allows for `max_length=65` characters it will require us to update the sdk.

Reply:
Yeah, not sure the best route there, we have similar issue with the password defaults https://github.com/clerk/javascript/blob/f9059a4ab24b39bc10041d7c37fbc73127e28817/packages/clerk-js/src/core/resources/UserSettings.ts#L19-L20

Reply:
since this is a pattern let's respect it